internal/engine/wazevo/backend/machine.go

```diff
@@ -61,6 +61,12 @@ type (
 		// ResolveRelocations resolves the relocations after emitting machine code.
 		ResolveRelocations(refToBinaryOffset map[ssa.FuncRef]int, binary []byte, relocations []RelocationInfo)
 
+		// UpdateRelocationInfo recomputes the RelocationInfo after emitting machine code and returns the updated
+		// RelocationInfo and the size of the trampoline if necessary.
+		UpdateRelocationInfo(refToBinaryOffset map[ssa.FuncRef]int, trampolineOffset int, r RelocationInfo) (res RelocationInfo, trampolineSize int)
```

Comment:
nit: Make this accept pointer and let's fix the callsite to avoid the heap allocation (see various places in the codebase)

```suggestion
		UpdateRelocationInfo(refToBinaryOffset map[ssa.FuncRef]int, trampolineOffset int, r *RelocationInfo) trampolineSize int
```

Reply:
I think it should be good now